lib/Viewer.js

```diff
@@ -185,16 +187,51 @@ Viewer.prototype.importXML = function(xml, done) {
       return done(err, parseWarnings);
     }
 
+    var diagram = null;
+    if (diagramId) {
+      diagram = find(definitions.diagrams, function(element) {
+        return element.id == diagramId;
+      });
+    }
+
     self.importDefinitions(definitions, function(err, importWarnings) {
       var allWarnings = [].concat(parseWarnings, importWarnings || []);
 
       self._emit('import.done', { error: err, warnings: allWarnings });
 
       done(err, allWarnings);
-    });
+    }, diagram);
   });
 };
 
+/**
+ * Switch to another diagram of the same XML
+ *
+ * @param {String} ID of the diagram to render (if null or non-existent, the first will be rendered)
+ * @param {Function} [done] invoked with (err, warnings=[])
+ */
+Viewer.prototype.switchDiagram = function(diagramId, done) {
+
+  // done is optional
+  done = done || function() {};
+
+  var self = this;
+
+  var diagram = null;
+  if (diagramId) {
+    diagram = find(self._definitions.diagrams, function(element) {
+      return element.id == diagramId;
+    });
+  }
+
+  self.importDefinitions(self._definitions, function(err, importWarnings) {
+
+    self._emit('import.done', { error: err, warnings: importWarnings });
```

Comment:
Diagram switch is not an import so it should not fire the `import.done` event. Let's use the render events only for such a case.